### PR TITLE
smc handler with sub id

### PR DESCRIPTION
**Pull request by rdevarajantct**
sub id added as part of the smc id in register a0.

### CODE REVIEW

File: core/arch/arm/kernel/thread_optee_smc.c [hunk 1]
```diff
@@ -282,9 +283,10 @@ static uint32_t std_smc_entry(uint32_t a0, uint32_t a1, uint32_t a2,
 					   with_rpc_arg);
 	case OPTEE_SMC_CALL_WITH_REGD_ARG:
 		return std_entry_with_regd_arg(reg_pair_to_64(a1, a2), a3);
-	case TRUSTCORE_SMC_CALL_WITH_REGD_ARG:
-		return trustcore_smc_handler(a1, a2, a3);
 	default:
+                if((a0 & (~0xFFu)) == TRUSTCORE_SMC_CALL_WITH_REGD_ARG) {
```
Comment (rneiman) [inline core/arch/arm/kernel/thread_optee_smc.c]: is there a macro for extracting the smsc id portion of the argument? if not why not add one?

Comment (rdevarajantct) [reply core/arch/arm/kernel/thread_optee_smc.c]: i am using the lower portion of the smc id (i.e. bits 0 to 8) for sending this sub ids. I didn't want to use a whole register to send this thats why its piggy backed on the smc id itself.

**Review by rneiman — COMMENTED** [review]
please see my comment

File: core/arch/arm/kernel/thread_optee_smc.c (hunk 1 above)
Comment (rneiman) [reply core/arch/arm/kernel/thread_optee_smc.c]: right - please provide aa macro that handles the '&' manipulation to make it more visual

**Review by rneiman — APPROVED** [review]

**Pull request merged**